### PR TITLE
Disable docs builds for dependabot

### LINKED ISSUE #101: Avoid deploying documentation with dependabot changes
We don't need to build or deploy docs for changes applied by dependabot. Additionally, there seem to be errors when it happens (potentially related to git history and dependabot's access privs).

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This change disables docs builds for changes related to dependabot (dependency updates).

Closes #101 

## What kind of change(s) are included?

- [x] Documentation (changes docs or other related content)
- [x] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [x] I have added comments to my code to help provide understanding
- [ ] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ on:
 
 jobs:
   build:
+    # only build and deploy docs if the actor is not dependabot
+    if: ${{ github.actor != 'dependabot[bot]' }}
```
Comment (jenna-tomkinson) [inline .github/workflows/publish-docs.yml:16]: What does actor mean in this context? What would be the actor if it isn't dependabot?

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! Nice short change! I only had one question so I can be on the same page lol.

File: .github/workflows/publish-docs.yml (hunk 1 above)
Comment (d33bs) [reply .github/workflows/publish-docs.yml:16]: Thanks for this question - the actor for GitHub actions is the latest person to trigger a workflow. That could be one of us or also automated agents like Dependabot, which act on our / GitHub's behalf to perform work.

**Comment by d33bs** [comment]
Thanks @jenna-tomkinson, merging this in!

**Pull request merged**